blocks/record.go

```diff
@@ -0,0 +1,46 @@
+package blocks
+
+import (
+	"io"
+
+	"github.com/bsm/go-sparkey"
+)
+
+// A Record is one key/value pair loaded from a block.
+type Record struct {
+	ValueLen uint64
+
+	iterPool iterPool
+	iter     *sparkey.HashIter
+	io.Reader
```

Comment:
Yeah—if you know you always have a combination `Reader`/`WriterTo` you can proxy both methods back by hand or similar